src/analyses/termination_new.ml

```diff
@@ -0,0 +1,116 @@
+(** Work in progress *)
```

Comment:
How about naming this something like `terminationLoops` or `loopTermination`?

Reply:
Done!